Language/Expressions/Property_Extraction/Generic_Method_Instantiation/generic_method_A01_t02.dart

```diff
@@ -20,22 +20,13 @@
 
 import '../../../../Utils/expect.dart';
 
-class C1 {
-  X foo<X extends num>(X x) => x;
-}
-
-class C2<T extends num> {
+class C<T extends num> {
   X foo<X extends T>(X x) => x;
 }
 
 void main() {
-  C1 c1 = C1();
-  Expect.throws(() {
-    String Function(int) f1 = c1.foo as dynamic;
-  });
-
-  C2 c2 = C2();
+  var c = C<num>();
   Expect.throws(() {
-    int Function(num) f2 = c2.foo as dynamic;
+    int Function(num) f = c.foo as dynamic;
```

Comment:
These tests are about generic function instantiation, so there should be a part of the code where this operation occurs. However, I can't see any such location in this test. In particular, `c.foo as dynamic` gives `c.foo` the context type `_`, which means that there will not be a generic function instantiation.

If we want to test the case where generic function instantiation occurs because of the failing caller-side check that I mentioned this morning then we'd need to satisfy two requirements:

- `c` must have a covariant type (as in `C<num> c = C<int>();`.
- An attempt is made to access `c.foo`.

This should be enough to make the program throw.

However, the thing that we should test is that this dynamic type check (that fails) is actually performed also in the case where the context calls for a generic function instantiation (which should be flawless according to the static types, such that we don't get any errors because of the function instantiation itself, it must throw simply because we're accessing `c.foo`).

Something like this:

```suggestion
    num Function(num) f = c.foo;
```

Reply:
Thanks!